manifests/init.pp

```diff
@@ -415,9 +417,10 @@
   Enum['ipmitool', 'freeipmi', 'shell'] $bmc_default_provider = $::foreman_proxy::params::bmc_default_provider,
   Boolean $realm = $::foreman_proxy::params::realm,
   Foreman_proxy::ListenOn $realm_listen_on = $::foreman_proxy::params::realm_listen_on,
-  String $realm_provider = $::foreman_proxy::params::realm_provider,
+  Enum['freeipa', 'ad'] $realm_provider = $::foreman_proxy::params::realm_provider,
   Stdlib::Absolutepath $realm_keytab = $::foreman_proxy::params::realm_keytab,
   String $realm_principal = $::foreman_proxy::params::realm_principal,
+  Optional[Foreman_proxy::AdConfig] $ad_config = $::foreman_proxy::params::ad_config,
```

Comment:
We still have an issue in our installer with a Struct because of Kafo. https://github.com/theforeman/puppet-foreman/pull/601 was limited by the same issue. I still need to look into this.